### PR TITLE
Add trigger to com_content front category model

### PR DESCRIPTION
**Pull request by Arkadiy-Sedelnikov**
Trigger a need for content filtering.

### CODE REVIEW

**Comment by roland-d** [comment]
Hello @Arkadiy-Sedelnikov 

Thank you for your contribution.

The last comment here was on 25th November. So the question is, Is this issue/pull request still valid? Does the PR linked by Bakual solve the issue?

If no reply is received within 4 weeks we will close this issue.

Thanks for understanding! <hr /><sub>This comment was created with the <a href="https://github.com/joomla/jissues">J!Tracker Application</a> at <a href="http://issues.joomla.org/tracker/joomla-cms/4656">issues.joomla.org/joomla-cms/4656</a>.</sub>

**Comment by Arkadiy-Sedelnikov** [comment]
I already solved the problem of the absence of the trigger in my extension substitution model. All it'd be better if the trigger was in the home model, but the urgent need for this anymore.

File: components/com_content/models/category.php [hunk 1]
```diff
@@ -239,6 +239,12 @@ function getItems()
 			$model->setState('filter.max_category_levels', $this->getState('filter.max_category_levels'));
 			$model->setState('list.links', $this->getState('list.links'));
 
+			$dispatcher = JEventDispatcher::getInstance();
+			// Include the content plugins for the change of category state event
```
Comment (roland-d) [inline components/com_content/models/category.php]: Please fix the comment as it is not a change of category state event.

File: components/com_content/models/category.php [hunk 2]
```diff
@@ -239,6 +239,12 @@ function getItems()
...
+			JPluginHelper::importPlugin('content');
+			// Trigger the onCategoryChangeState event
```
Comment (roland-d) [inline components/com_content/models/category.php]: Please fix the comment as it is not a change of category state event.

File: components/com_content/models/category.php [hunk 3]
```diff
@@ -239,6 +239,12 @@ function getItems()
...
+			$dispatcher->trigger('onGetContentItems', array(&$model));
```
Comment (Bakual) [inline components/com_content/models/category.php]: You don't need to fetch the dispatcher as an own object and trigger it. Just use `JFactory::getApplication()->triggerEvent('onGetContentItems', array(&$model)))`

**Comment by mbabker** [comment]
I'm confused on what exactly this is supposed to be doing.

**Comment by FPerisa** [comment]
Testing the trigger was successful. The trigger calls the function "onGetContentItems" for a content plugin like "contact".

**Comment by Arkadiy-Sedelnikov** [comment]
See http://demoj3.argens.ru/index.php/blog  
Module "MiniCCK Filter" working on this trigger.

**Comment by mbabker** [comment]
I get that the trigger is added and running.  What I don't get is why it's being added and what the purpose is for this very specific use case.  Generally, we do not have event triggers to manipulate an object (model or database query typically) before a data query is executed; this changes that behavior for one very exclusive use case.  Right now this PR looks to me like a "I need this feature for my extension so I'm going to add it without anymore detail" situation and my inclination is to suggest it be closed because of that (looks to be something for an edge case in an extension and no clarifying explanation about why this is proposed).  IMO, this PR needs to be better explained; why is this event trigger needed for only this one use case out of all the components shipped with the Joomla core.

**Comment by Arkadiy-Sedelnikov** [comment]
I wrote that for my decisions, this PR is not mandatory, I already made a spoof of this model with a built-in trigger. The trigger I use for content filtering ин additional fields on my CCK, so it can use other CCK parasitic on com_content.

**Comment by Bakual** [comment]
I'm closing this PR as in the current state it will not get merged. To many questions are open.

If we want to introduce this new plugin event, it should better be placed in a library model somewhere so it's triggered for all extensions, not a single one.

**Pull request closed without merging**